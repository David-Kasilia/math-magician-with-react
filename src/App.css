*,
*::before,
*::after {
  box-sizing: border-box;
}

body {
  margin: 0;
}

.calculator-grid {
  display: grid;
  margin-top: 2rem;
  position: fixed;
  left: 25%;
  right: 20%;
  top: 10%;
  grid-template-columns: repeat(8, 6rem);
  /* stylelint-disable-next-line csstree/validator */
  grid-template-rows: minmax(7rem, auto) repeat(5 6rem);
}

.operand {
  grid-column: 1 / -1;
  background-color: #858694;
  height: 10vh;
  margin-top: 10px;
  color: white;
  text-align: right;
  font-size: 42px;
  font-family: 'Times New Roman', Times, serif;
}

.btn {
  height: 5rem;
  font-size: 20px;
  text-align: center;
  grid-column: span 2;
}

.btn1 {
  height: 5rem;
  font-size: 20px;
  text-align: center;
  background-color: #f5913e;
  grid-column: span 2;
}

.btn-equal,
.btn-divide,
.btn-subtract,
.btn-add,
.btn-multiply {
  background: #f5913e;
}

.btn-zero {
  grid-column: 1 / 5;
}
