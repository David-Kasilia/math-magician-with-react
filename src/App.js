import React from 'react';
import './App.css';
import Calculator from './components/Calculator';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      data: {
        total: null,
        next: null,
        operation: null,
      },
    };
  }

  render() {
    const { data } = this.state;
    return (
      <Calculator data={data} />
    );
  }
}

export default App;
