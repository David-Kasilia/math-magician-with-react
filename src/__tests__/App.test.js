import { render, screen } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import App from '../App';

test('Unit Test Check For App Component', () => {
  render(<App />);
  const linkElement = screen.getByText(/Welcome To Our Page/i);
  expect(linkElement).toBeInTheDocument();
});

describe('Unit Test For  App component', () => {
  it('App Component renders correctly', () => {
    const tree = TestRenderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
