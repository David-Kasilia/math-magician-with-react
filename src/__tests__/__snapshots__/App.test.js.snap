// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Test For  App component App Component renders correctly 1`] = `
Array [
  <header
    className="header"
  >
    <h1
      className="title"
    >
      Math Magicians
    </h1>
    <nav
      className="nav-Bar"
    >
      <ul
        className="nav-items"
      >
        <li
          className="nav-item link"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="line"
        >
          |
        </li>
        <li
          className="nav-item link"
        >
          <a
            href="/calculator"
            onClick={[Function]}
          >
            Calculator
          </a>
        </li>
        <li
          className="line"
        >
          |
        </li>
        <li
          className="nav-item link"
        >
          <a
            href="/quote"
            onClick={[Function]}
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <div
    className="home-container"
  >
    <h2
      className="home-title"
    >
      Welcome To Our Page
    </h2>
    <p
      className="home-text"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </p>
    <p
      className="home-text"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </p>
  </div>,
]
`;
